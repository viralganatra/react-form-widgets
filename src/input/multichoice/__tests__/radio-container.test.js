import React from 'react';
import RadioContainer from '../radio-container';
import GroupComponent from '../__mocks__/component-group';

jest.mock('../../../events');

const props = {
    items: [
        { label: 'test 1', value: '1' },
        { label: 'test 2', value: '2' },
    ],
    value: '2',
};

describe('RadioContainer', () => {
    let Composed;

    beforeAll(() => {
        Composed = RadioContainer(GroupComponent);
    });

    it('should render correctly', () => {
        const wrapper = shallow(<Composed {...props} />);

        expect(() => RadioContainer({})).toThrowErrorMatchingSnapshot();
        expect(wrapper).toMatchSnapshot();
    });

    it('should validate prop types', () => {
        const wrapperInvalidItems = () => shallow(
            <Composed {...props} items={2} />,
        );

        const wrapperInvalidType = () => shallow(
            <Composed {...props} type="checkbox" />,
        );

        const wrapperInvalidValue = () => shallow(
            <Composed {...props} value={2} />,
        );

        expect(wrapperInvalidItems).toThrowErrorMatchingSnapshot();
        expect(wrapperInvalidType).toThrowErrorMatchingSnapshot();
        expect(wrapperInvalidValue).toThrowErrorMatchingSnapshot();
    });

    it('should listen to the onChange event and return the checked value', () => {
        const methodSpy = jest.fn();
        const wrapper = mount(<Composed {...props} onChange={methodSpy} />);

        wrapper.find('input').at(0).simulate('change', { target: { value: '1' } });

        expect(wrapper).toMatchSnapshot();

        wrapper.find('input').at(1).simulate('change', { target: { value: '2' } });

        expect(methodSpy.mock.calls[0][0].value).toBe('1');
        expect(methodSpy.mock.calls[1][0].value).toBe('2');

        expect(wrapper).toMatchSnapshot();
    });
});
