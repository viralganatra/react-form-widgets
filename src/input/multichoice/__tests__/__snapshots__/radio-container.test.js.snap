exports[`RadioContainer should listen to the onChange event and return the checked value 1`] = `
<RadioContainer
  items={
    Array [
      Object {
        "label": "test 1",
        "value": "1",
      },
      Object {
        "label": "test 2",
        "value": "2",
      },
    ]
  }
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onKeyUp={[Function]}
  type="radio"
  value="2">
  <GroupComponent
    isChecked={[Function]}
    items={
      Array [
        Object {
          "label": "test 1",
          "value": "1",
        },
        Object {
          "label": "test 2",
          "value": "2",
        },
      ]
    }
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onKeyUp={[Function]}
    type="radio">
    <div>
      <input
        checked={true}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        type="radio"
        value="1" />
      <input
        checked={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        type="radio"
        value="2" />
    </div>
  </GroupComponent>
</RadioContainer>
`;

exports[`RadioContainer should listen to the onChange event and return the checked value 2`] = `
<RadioContainer
  items={
    Array [
      Object {
        "label": "test 1",
        "value": "1",
      },
      Object {
        "label": "test 2",
        "value": "2",
      },
    ]
  }
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onKeyUp={[Function]}
  type="radio"
  value="2">
  <GroupComponent
    isChecked={[Function]}
    items={
      Array [
        Object {
          "label": "test 1",
          "value": "1",
        },
        Object {
          "label": "test 2",
          "value": "2",
        },
      ]
    }
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onKeyUp={[Function]}
    type="radio">
    <div>
      <input
        checked={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        type="radio"
        value="1" />
      <input
        checked={true}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        type="radio"
        value="2" />
    </div>
  </GroupComponent>
</RadioContainer>
`;

exports[`RadioContainer should render correctly 1`] = `
"Expected \"WrappedComponent\" provided as the first argument to
        RadioContainerHOC to be a function. Instead, received
        object."
`;

exports[`RadioContainer should render correctly 2`] = `
<GroupComponent
  isChecked={[Function]}
  items={
    Array [
      Object {
        "label": "test 1",
        "value": "1",
      },
      Object {
        "label": "test 2",
        "value": "2",
      },
    ]
  }
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onKeyUp={[Function]}
  type="radio" />
`;

exports[`RadioContainer should validate prop types 1`] = `
"Warning: Failed prop type: Invalid prop \`items\` of type \`number\` supplied to \`RadioContainer\`, expected an array.
    in RadioContainer"
`;

exports[`RadioContainer should validate prop types 2`] = `
"Warning: Failed prop type: Invalid prop \`type\` of value \`checkbox\` supplied to \`RadioContainer\`, expected one of [\"radio\"].
    in RadioContainer"
`;

exports[`RadioContainer should validate prop types 3`] = `
"Warning: Failed prop type: Invalid prop \`value\` of type \`number\` supplied to \`RadioContainer\`, expected \`string\`.
    in RadioContainer"
`;
